src/humanize/number.py

```diff
@@ -516,7 +516,12 @@ def metric(value: float, unit: str = "", precision: int = 3) -> str:
     Returns:
         str:
     """
-    exponent = int(math.floor(math.log10(abs(value)))) if value != 0 else 0
+    if math.isnan(value):
+        return "NaN"
+    elif value != 0:
+        exponent = int(math.floor(math.log10(abs(value))))
+    else:
+        exponent = 0
```

Comment:
Don't edit the existing L519, just add the new conditional. These two conditionals are not related, so they shouldn't be linked together with elif.

Also, we should be as consistent with the existing behavior as possible, so we should return "NaN m" if the unit is meters? I'm not sure about the details here because I haven't used this function a lot.

```suggestion
    if math.isnan(value):
        return f"{value} {unit}"
    exponent = int(math.floor(math.log10(abs(value)))) if value != 0 else 0
```

Reply:
OK so if I add the condition for isfinite() will it be merged?